core/dbt/parser/manifest.py

```diff
@@ -1291,9 +1303,12 @@ def _process_refs_for_metric(manifest: Manifest, current_project: str, metric: P
 
         metric.depends_on.nodes.append(target_model_id)
 
-        _process_semantic_information_for_metric(
-            manifest, target_model, metric, target_model_package, current_project
-        )
+        if target_model.is_public:
```

Comment:
Normally we want to limit the post-processing things we do in parsing as much as possible, to only those things that CAN'T be done earlier. Is there some reason that the process_semantic_information_for_metric and  _validate_non_public_metric things aren't being done when nodes are normally validated?

Reply:
When you say `normally validated` are you talking about the classes in `unparsed.py` and `parsed.py`?

Reply:
Yes